### PR TITLE
Migrated MacOS keychain construct-based plugin to use dtfabric #1893

### PR DESCRIPTION
**Pull request by joachimmetz**
Migrated MacOS keychain construct-based plugin to use dtfabric #1893

### CODE REVIEW

File: plaso/parsers/mac_keychain.py [hunk 1]
```diff
@@ -183,298 +142,719 @@ class KeychainParser(interface.FileObjectParser):
       'imap': 'imap',
       'http': 'http'}
 
-  def _ReadEntryApplication(self, parser_mediator, file_object):
-    """Extracts the information from an application password entry.
+  # TODO
```
Comment (Onager) [inline plaso/parsers/mac_keychain.py]: Todo what? ;)

File: plaso/parsers/mac_keychain.py [hunk 2]
```diff
@@ -500,46 +879,21 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
...
-      # Table_offset: absolute byte in the file where the table starts.
-      # table.first_record: first record in the table, relative to the
-      #                     first byte of the table.
-      file_object.seek(table_offset + table.first_record, os.SEEK_SET)
+    tables = self._ReadTablesArray(file_object, file_header.tables_array_offset)
 
-      if table.record_type == self.RECORD_TYPE_INTERNET:
-        for _ in range(table.number_of_records):
-          self._ReadEntryInternet(parser_mediator, file_object)
+    table = tables.get(self._RECORD_TYPE_APPLICATION_PASSWORD, None)
```
Comment (Onager) [inline plaso/parsers/mac_keychain.py]: You're calling get() with an int argument here, but the docstring from _ReadTablesArray indicates that the dict has string keys. Please check the docstring for the function, I can't readily tell what it's supposed to be.

File: plaso/parsers/mac_keychain.py [hunk 3]
```diff
@@ -183,298 +142,719 @@ class KeychainParser(interface.FileObjectParser):
...
+      raise errors.ParseError('Unsupported format version: {0:s}.{1:s}'.format(
+          file_header.major_format_version, file_header.minor_format_version))
+
+    return file_header
+
+  def _ReadRecord(self, tables, file_object, record_offset, record_type):
+    """Reads the record.
+
+    Args:
+      tables (dict[str, KeychainDatabaseTable]): tables per name.
```
Comment (Onager) [inline plaso/parsers/mac_keychain.py]: Should be 'str' be 'int' here? On line 363, you use an int as a key into this dict.

File: plaso/parsers/mac_keychain.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/parsers/mac_keychain.py]: this was/is my refactor marker, I've removed it.

File: plaso/parsers/mac_keychain.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/mac_keychain.py]: made some changes.

File: plaso/parsers/mac_keychain.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/parsers/mac_keychain.py]: made some changes.

**Pull request merged**